### ISSUE TITLE
Add: logging... 

### ISSUE THREAD

**Issue by benrussill** [BunnyWay/BunnyCDN.Net.Storage#17]
Seriously. Add logging. 

[Stop Using String Interpolation When Logging in .NET (Nick Chapsas)](https://youtu.be/6zoMd_FwSwQ) 

[High Performance Logging in .NET, the proper way (Nick Chapsas)](https://youtu.be/a26zu-pyEyg) 

[You're doing .NET logging wrong. Let's fix it. (Nick Chapsas)](https://youtu.be/bnVfrd3lRv8)

**Comment by OhSoooLucky** [comment]
I do not believe logging in a library should be a thing. This is up to the application builder that uses the library to implement and handle errors thrown by the library
